homeassistant/components/modbus/__init__.py

```diff
@@ -292,6 +294,8 @@
         vol.Optional(CONF_TIMEOUT, default=3): cv.socket_timeout,
         vol.Optional(CONF_CLOSE_COMM_ON_ERROR, default=True): cv.boolean,
         vol.Optional(CONF_DELAY, default=0): cv.positive_int,
+        vol.Optional(CONF_RETRIES, default=3): cv.positive_int,
+        vol.Optional(CONF_RETRY_ON_EMPTY, default=False): cv.boolean,
```

Comment:
Are these options only applicable to the serial client or to all client types?

Reply:
Right now I have only activated it for serial, because I had a user that was lost without this change. Next week I will test it on the other communications as well.

I could have added it in the serial schema, but then I would have to change the schema next week, I preferred to have this general….test the other communications and add it to them. Anyhow this PR will not be released before 2021.7 so there are plenty of time.

If you insist I can add parameters to other communications in this PR, the parameters are allowed but I have not tested the implementation.

Reply:
The config validation should be as precise as possible. Please adjust it.

Reply:
Adjusted